meteor/sofie-server/model/entities/part.ts

```diff
@@ -62,32 +97,99 @@ export class Part {
 		return this.isPartNext
 	}
 
-	public getTimelineObjects(): TimelineObject[] {
-		const now: number = new Date().getTime()
-		const adLibTimelineObjects: TimelineObject[] = this.adLibPieces
-			.filter((piece) => this.shouldAdLibPieceBeShown(piece, now))
-			.flatMap((piece) => piece.timelineObjects)
-		const pieceTimelineObjects: TimelineObject[] = this.pieces.flatMap((piece) => piece.timelineObjects)
-		return adLibTimelineObjects.concat(pieceTimelineObjects)
+	public getPieces(): Piece[] {
+		return this.pieces
 	}
 
-	private shouldAdLibPieceBeShown(adLibPiece: AdLibPiece, executionTime: number): boolean {
-		return (
-			adLibPiece.getExecutedAt() > 0 &&
-			adLibPiece.getExecutedAt() <= executionTime &&
-			adLibPiece.getExecutedAt() + adLibPiece.duration > executionTime
-		)
+	public setPieces(pieces: Piece[]): void {
+		this.pieces = pieces
 	}
 
 	public addAdLibPiece(adLibPiece: AdLibPiece): void {
 		this.adLibPieces.push(adLibPiece)
 	}
 
-	public getInfiniteRundownPieces(): Piece[] {
-		return this.pieces.filter((piece) => piece.pieceLifeSpan === PieceLifeSpan.INFINITE_RUNDOWN)
+	public getPiecesWithLifespan(lifespanFilters: PieceLifespan[]): Piece[] {
+		return this.pieces.filter((piece) => lifespanFilters.includes(piece.pieceLifespan))
+	}
+
+	public getExecutedAt(): number {
+		return this.executedAt
+	}
+
+	// TODO: This implementation currently reflects how Core implemented it. It's in dire need of a refactor.
+	public calculateTimings(previousPart?: Part): void {
+		const maxPreRollDurationFromPieces: number = this.pieces
+			// Note: Core filters for !BlueprintPieceType.Normal and piece.enable.start !== 'now' - Will does Pieces ever have a PreRollDuration?
+			.reduce((preRollDuration: number, piece: Piece) => Math.max(preRollDuration, piece.preRollDuration ?? 0), 0)
+
+		const maxPostRollDurationForPieces: number = this.pieces
+			.filter((piece) => !!piece.postRollDuration && !piece.duration)
+			.reduce((postRollDuration: number, piece: Piece) => Math.max(postRollDuration, piece.postRollDuration), 0)
```

Comment:
Who ever does  the refactor of this, could/should move the overlap of the logic between these to a seperate reuseable method.
Following is just some notes/ideas for what i had in mind - might not compile.

``` typescript
public calculateTimings(previousPart?: Part): void {
   const maxPreRollDurationFromPieces: number = this.getDurationFromPieces(
      this.pieces,
      (piece: piece) => piece.preRollDuration
   )
   const maxPostRollDurationForPieces: number this.getDurationFromPieces(
      this.pieces.filter((piece) => !!piece.postRollDuration && !piece.duration), 
      (piece: piece) => piece.postRollDuration
   )
   ...
}

...
getDurationFromPieces(pieces: Pieces[], chosenNumber: (piece: Piece) => number): number {
   return pieces.reduce(duration: number, piece: Piece) => Math.max(duration, chosenNumber(piece) ?? 0), 0)
}
...
```

Reply:
The reduce is already extracted into a single reusable method: `Math.max()`.